src/main/java/org/candlepin/subscriptions/event/EventController.java

```diff
@@ -365,6 +370,47 @@ private void ensureOptIn(String orgId) {
     }
   }
 
+  private List<EventRecord> updateIngestedUsage(List<EventRecord> eventRecords) {
+    for (Event event :
+        eventRecords.stream().map(EventRecord::getEvent).collect(Collectors.toList())) {
+      if (event.getProductTag() == null
+          || event.getMeasurements() == null
+          || event.getMeasurements().size() == 0) {
+        continue;
+      }
+      if (event.getEventSource() != null && event.getEventSource().equalsIgnoreCase("prometheus")) {
+        for (String tag : event.getProductTag()) {
+          for (Measurement measurement : event.getMeasurements()) {
+            String billingProvider =
+                event.getBillingProvider() == null ? null : event.getBillingProvider().value();
+            if (billingProvider != null) {
+              meterRegistry
+                  .counter(
+                      "rhsm-subscriptions.swatch_metrics_ingested_usage_total",
+                      "product_tag",
+                      tag,
+                      "metric_id",
+                      measurement.getMetricId(),
+                      "billing_provider",
+                      billingProvider)
+                  .increment(measurement.getValue());
+            } else {
+              meterRegistry
+                  .counter(
+                      "rhsm-subscriptions.swatch_metrics_ingested_usage_total",
+                      "product_tag",
+                      tag,
+                      "metric_id",
+                      measurement.getMetricId())
+                  .increment(measurement.getValue());
+            }
```

Comment:
To avoid having the same counter twice and almost duplicated, this is my suggested change:

```java
List<String> tags = new ArrayList<>(List.of("product_tag", tag, "metric_id", measurement.getMetricId());
if (event.getBillingProvider() != null) {
  tags.addAll(List.of("billing_provider", event.getBillingProvider()));
}

meterRegistry
        .counter(
                "rhsm-subscriptions.swatch_metrics_ingested_usage_total",
                tags.toArray(new String[0]))
        .increment(measurement.getValue());
```

Basically, it populates a list with the tags and then creates the array from it which is what the `counter` needs.